appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/jmx/MBeanMetadataHelper.java

```diff
@@ -88,7 +88,7 @@
     public List<MBeanMetadata> registerMetadata(MetricRegistry metricRegistry,
             List<MBeanMetadata> metadataList, boolean isRetry) {
 
-        if (!metricRegistry.getMetadata().isEmpty() && !isRetry) {
+        if (!metricRegistry.getNames().isEmpty() && !isRetry) {
```

Comment:
NB. `getNames()` is a view, whereas `getMetadata()` is computed collection, therefore the change.